policies/branch-policy.md

```diff
@@ -0,0 +1,100 @@
+Branch Policy
+=============
+
+The openssl repository contains the following maintained branches:
+
+## The master branch
+
+- Any type (bug fix, feature, refactoring…) of pull requests is allowed.
+
+- The development of the next minor or major release happens there.
+  Which kind of release it is is defined by OMC at some point after the
+  master branch is opened after the previous release was branched.
+  Implicitly it means that any API/ABI breaking changes are not allowed
+  until OMC defines the intended kind (minor/major) of the next release.
+
+**If the next minor branch is approved the following applies.**
+
+- The development of the next major release happens there. Implicitly it
+  means that any API/ABI breaking changes are allowed but OMC can (and
+  usually will) limit the extent of the breakage allowed.
+- Major features are allowed. Examples of a major feature: A completely
+  new implementation of a protocol. New API for pluggable crypto modules.
+- Major refactoring is allowed. Examples of major refactoring: Splitting
+  libcrypto into multiple hierarchically dependent libraries.
+
+## The supported release branches
+
+- Only bug fixes and documentation changes are allowed.
+- The development of the next patch releases of supported minor releases
+  happens there.
+- By exception given by OMC also other types of pull requests can be merged.
+- During the final year of support, we do not commit to anything other than
+  security fixes.
+- Bug fix and documentation pull requests must be always merged to the
+  latest branch (including the master branch) where the bug or documentation
+  deficiency is present. It can be then merged (backported or directly
+  cherry-picked) to all older branches where the deficiency is present.
```

Comment:
Is there to be any distinction between release branches for LTS and non-LTS release branches?
(I personally am okay with no difference, but want to check if that is the intent.)

Reply:
In terms of what is allowed to commit to LTS and non-LTS release branches this policy does not distinguish them. We might modify the policy later if we find that LTS branches need some different treatment.